src/org/opensolaris/opengrok/search/context/Context.java

```diff
@@ -64,34 +73,164 @@
      * whose values tell if the field is case insensitive (true for
      * insensitivity, false for sensitivity).
      */
-    private static final Map<String, Boolean> tokenFields =
+    private static final Map<String, Boolean> TOKEN_FIELDS =
             new HashMap<String, Boolean>();
     static {
-        tokenFields.put("full", Boolean.TRUE);
-        tokenFields.put("refs", Boolean.FALSE);
-        tokenFields.put("defs", Boolean.FALSE);
+        TOKEN_FIELDS.put(QueryBuilder.FULL, Boolean.TRUE);
+        TOKEN_FIELDS.put(QueryBuilder.REFS, Boolean.FALSE);
+        TOKEN_FIELDS.put(QueryBuilder.DEFS, Boolean.FALSE);
     }
 
     /**
-     * Constructs a context generator
+     * Initializes a context generator for matchers derived from the specified
+     * {@code query} -- which might be {@code null} and result in
+     * {@link #isEmpty()} returning {@code true}.
      * @param query the query to generate the result for
-     * @param queryStrings map from field names to queries against the fields
+     * @param qbuilder required builder used to create {@code query}
      */
-    public Context(Query query, Map<String, String> queryStrings) {
+    public Context(Query query, QueryBuilder qbuilder) {
+        if (qbuilder == null) {
+            throw new IllegalArgumentException("`qbuilder' is null");
+        }
+
+        this.query = query;
+        this.qbuilder = qbuilder;
         QueryMatchers qm = new QueryMatchers();
-        m = qm.getMatchers(query, tokenFields);
+        m = qm.getMatchers(query, TOKEN_FIELDS);
         if (m != null) {
-            buildQueryAsURI(queryStrings);
+            buildQueryAsURI(qbuilder.getQueries());
             //System.err.println("Found Matchers = "+ m.length + " for " + query);
             buffer = new char[MAXFILEREAD];
             tokens = new PlainLineTokenizer((Reader) null);
         }
     }
 
+    /**
+     * Toggles the alternating value (initially {@code true}).
+     */
+    public void toggleAlt() {
+        alt = !alt;
+    }
+
     public boolean isEmpty() {
         return m == null;
     }
 
+    /**
+     * Look for context for this instance's initialized query in a search result
+     * {@link Document}, and output according to the parameters.
+     * @param env required environment
+     * @param searcher required search that produced the document
+     * @param docId document ID for producing context
+     * @param dest required target to write
+     * @param urlPrefix prefix for links
+     * @param morePrefix optional link to more... page
+     * @param limit a value indicating if the number of matching lines should be
+     * limited. N.b. unlike
+     * {@link #getContext(java.io.Reader, java.io.Writer, java.lang.String, java.lang.String, java.lang.String, org.opensolaris.opengrok.analysis.Definitions, boolean, boolean, java.util.List, org.opensolaris.opengrok.analysis.Scopes)},
```

Comment:
Maybe break the line so it won't be that long? I do not think that it will break the javadoc.